src/libraries/System.Text.RegularExpressions/tests/Regex.KnownPattern.Tests.cs

```diff
@@ -1117,6 +1115,95 @@ public async Task Docs_Anchors_ContiguousMatches(RegexEngine engine)
                 Regex.Replace(Input, Pattern, m => string.Concat(m.Value.Reverse())));
         }
 
+        //
+        // Based on examples from https://blog.stevenlevithan.com/archives/balancing-groups
+        //
+
+        [Theory]
+        [MemberData(nameof(RegexHelpers.AvailableEngines_MemberData), MemberType = typeof(RegexHelpers))]
+        public async Task Blog_Levithan_BalancingGroups_Palindromes(RegexEngine engine)
+        {
+            if (RegexHelpers.IsNonBacktracking(engine))
+            {
+                // balancing groups not supported
+                return;
+            }
+
+            Regex r = await RegexHelpers.GetRegexAsync(engine, @"(?<N>.)+.?(?<-N>\k<N>)+(?(N)(?!))");
+
+            // Palindromes
+            Assert.All(new[]
+            {
+                "kayak",
+                "racecar",
+                "never odd or even",
+                "madam im adam"
+            }, p => Assert.True(r.IsMatch(p)));
+
+            // Non-Palindromes
+            Assert.All(new[]
+            {
+                "canoe",
+                "raceboat"
+            }, p => Assert.False(r.IsMatch(p)));
+        }
+
+        [Theory]
+        [MemberData(nameof(RegexHelpers.AvailableEngines_MemberData), MemberType = typeof(RegexHelpers))]
+        public async Task Blog_Levithan_BalancingGroups_MatchingParentheses(RegexEngine engine)
+        {
+            if (RegexHelpers.IsNonBacktracking(engine))
+            {
+                // balancing groups not supported
+                return;
+            }
+
+            Regex r = await RegexHelpers.GetRegexAsync(engine, @"^\(
+	                                                                 (?>
```

Comment:
NIT: looks like both of the Regexes you added (the on on this test and the next one) have tabs in them.

Reply:
Oops, thanks.